sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/rest/SwaggerMethodParser.java

```diff
@@ -603,4 +614,22 @@ private Map<Integer, UnexpectedExceptionInformation> processUnexpectedResponseEx
 
         return exceptionHashMap;
     }
+
+    /**
+     * Indicates whether the swagger method is of Reactive type or not.
+     * To be classified as reactive method, either the return type or one of the input parameters must
+     * be one of {@link Mono}, {@link Flux} or {@link Publisher}.
+     *
+     * @return the boolean flag indicating whether the swagger method is reactive or not.
+     */
+    public boolean isReactive() {
+        return isReactive;
+    }
+
+    boolean isReactiveType(Type type) {
+        return (TypeUtil.isTypeOrSubTypeOf(type, Mono.class) || TypeUtil.isTypeOrSubTypeOf(type, Flux.class)
+            || TypeUtil.isTypeOrSubTypeOf(type, Publisher.class)
+            || TypeUtil.isTypeOrSubTypeOf(type, StreamResponse.class));
+        //Note: StreamResponse will be added to sync flows in a follow-up PR. Currently, their reactive body types are handled in Async Rest Proxy.
```

Comment:
This is hacky. I suggest we add `isStreamResponse` and use it in the place we fork to sync/async proxy.
StreamResponse is neither reactive nor non-reactive. It depends on how it's consumed later. It doesn't fit here.